tools/samtools.wdl

```diff
@@ -814,19 +887,25 @@ task fixmate {
         extension: {
             description: "File format extension to use for output file.",
             choices: [
-                ".sam",
+                ".sam",  # TODO are we sure we want to encourage the use of .sam?
```

Comment:
I think it's more a question of fully supporting the underlying tool options. It is also possible that in the future, you could stream output between tasks, rather than passing files around (maybe?). SAM may make sense in that application.

Reply:
Fair point on the future application. I hadn't thought of that, but skipping [de]compression makes sense in that case.

> I think it's more a question of fully supporting the underlying tool options

I disagree that this is necessary. There's precedence for us saying "we're not exposing XYZ parameter because it's silly/bad/etc." We've done it with STAR, Qualimap RNAseq, probably others I'm blanking on at the moment.

As I see it, there are 2 philosophies conflicting here:
1) we should expose all parameters we can. **Trust the end user to not break things.**
2) **we should provide a curated set of parameters to users** (because we can't really trust the end user to not break things)
    * In most cases that is "all parameters" and this looks like philosophy (1)
    * in certain cases we should hide "bad" options from users. Defining "bad" is tough, but I would call "passing around uncompressed SAM data and storing it on disk at least once if not more times" an objectively "bad" idea.
    * As I/O streaming is not yet a feature of WDL, I do not see _any_ case where SAM output is appropriate.

I vote we remove `".sam"` from `choices` and call it a day. A user could still pass in `.sam` and get the "desired" behavior, but we don't broadcast it as a possibility. If streaming is added in the future, we revisit the best way to expose this.

Reply:
I'm not sure I agree on SAM being bad in this context. The default is BAM. If a user wants SAM, it's up to them to understand the storage issue. I don't have a concrete idea for why someone would want to do that, but they might. I suppose removing it from `choices` would be fine. It doesn't remove the functionality, but does make it a little more hidden.